### PR TITLE
Custom base url (LLC)

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

File: protocol-tests/src/main/java/module-info.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
-module fixtures.bodystring {
```
Comment (weidongxu-microsoft) [inline protocol-tests/src/main/java/module-info.java:1]: Should we delete the module-info file in `generate` script?

In my case in fluent java8 not able to recognize it.
https://github.com/Azure/autorest.java/blob/v4/fluent-tests/prepare-tests.bat#L48-L49

File: javagen/src/main/java/com/azure/autorest/model/clientmodel/ClassType.java [hunk 2]
```diff
@@ -74,7 +74,7 @@
     public static final ClassType JsonPatchDocument =
             new ClassType.Builder().knownClass(com.azure.core.models.JsonPatchDocument.class).build();
     public static final ClassType BinaryData = new ClassType.Builder().knownClass(com.azure.core.util.BinaryData.class).defaultValueExpressionConverter((String defaultValueExpression) -> java.lang.String.format("BinaryData.fromObject(\"%s\")", defaultValueExpression)).build();
-    public static final ClassType RequestOptions = new Builder().packageName("com.azure.core.http").name("RequestOptions").build();
+    public static final ClassType RequestOptions = new Builder().packageName("com.azure.core.http.rest").name("RequestOptions").build();
```
Comment (qwordy) [inline javagen/src/main/java/com/azure/autorest/model/clientmodel/ClassType.java:77]: Is `RequestOptions` in `com.azure.core.http` deprecated?

File: javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java [hunk 3]
```diff
@@ -273,8 +273,8 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
         if (settings.isLowLevelClient()) {
             imports.add("com.azure.core.http.HttpMethod");
             imports.add("com.azure.core.http.rest.Response");
-            imports.add("com.azure.core.http.RequestOptions");
-            imports.add("com.azure.core.util.BinaryData");
+            ClassType.RequestOptions.addImportsTo(imports, true);
+            ClassType.BinaryData.addImportsTo(imports, true);
```
Comment (qwordy) [inline javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java:277]: What's the difference? `implementationImports` in builder is never used in project.

**Review by qwordy — APPROVED** [review]

**Comment by qwordy** [comment]
@jianghaolu, if there is no problem, we can merge it. Conflict has appeared since llc branch is changing fast.

**Pull request closed without merging**